src/osal/lv_freertos.c

```diff
@@ -261,8 +272,19 @@ lv_result_t lv_thread_sync_wait(lv_thread_sync_t * pxCond)
 lv_result_t lv_thread_sync_signal(lv_thread_sync_t * pxCond)
 {
 #if USE_FREERTOS_TASK_NOTIFY
-    /* Send a notification to the task waiting. */
-    xTaskNotifyGive(pxCond->xTaskToNotify);
+    _enter_critical();
+    TaskHandle_t task_to_notify = pxCond->xTaskToNotify;
```

Comment:
Use the new member `BaseType_t    xIsInitialized; ` to verify if the task to notify is valid.